Sources/SwiftParser/Statements.swift

```diff
@@ -1106,10 +1106,23 @@ extension Parser.Lookahead {
     case .identifier:
       // "identifier ':' for/while/do/switch" is a label on a loop/switch.
       guard self.peek().tokenKind == .colon else {
-        // "yield" in the right context begins a yield statement.
-        if context == .coroutineAccessor, self.atContextualKeyword("yield") {
-          return true
+        if self.atContextualKeyword("yield") {
+          switch self.peek().tokenKind {
+          case .prefixAmpersand:
+            // "yield &" always denotes a yield statement.
+            return true
+          case .leftParen:
+            // "yield (", by contrast, must be disambiguated with additional
+            // context. We always consider it an apply expression of a function
+            // called `yield` for the purposes of the parse.
+            return false
+          default:
+            // "yield" followed by any other token is likely a yield statement
+            // of some singular expression.
+            return true
```

Comment:
Hmm. Should we check whether the next token is on a new line?

```swift
x = yield
print("huh")
```

?

Reply:
Ooh, good catch.

Reply:
```
yield
print("huh")
```